### PR TITLE
DEV-2530: Fix bug with fees at page load

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

This fixes a bug with how fee calculation works on a contribution page at pageload. This bug only affects older contribution pages that have contribution amounts stored as strings.

I refactored out two contribution amount-related utility functions into a separate module to help with testability. I tried leaving them in the `<DAmount>` component but Jest does not like importing a module named `constants` because our own module conflicts with one in node_modules, and this also helps start to detangle some circular dependencies we have in our code (I think).

I also fixed the `npm run test:coverage` command so it outputs accurate info.

#### Why are we doing this? How does it help us?

Fixes a bug with older donation pages.

#### How should this be manually tested? Please include detailed step-by-step instructions.

Repeat the steps below with the following contribution pages. I looked in the Django admin to find an example of a page that uses quoted values vs. one that uses numbers.
- https://newsrevenuehub-dev-2530.revengine-review.org/donate (old, quoted values)
- https://shawarma-dev-2530.revengine-review.org/donate (new, unquoted values)

1. Go to the contribution page.
2. Confirm that a button is selected in the contribution amounts.
3. Confirm that the Stripe fee looks accurate (e.g. not super high).
4. Confirm that the submit button at the bottom of the page shows the correct amount.
5. Fill out the rest of the fields and confirm that the payment amount requested is correct.
6. Complete steps 1-5 but choose a recurring contribution.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No.

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

n/a

#### Are there any smells or added technical debt to note?

No.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-2530

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

Subtasked https://news-revenue-hub.atlassian.net/browse/DEV-2584 for database cleanup.

### CODE REVIEW

**Review by benjaminEwhite — APPROVED** [review]
No complaints here. LGTM!

File: spa/src/components/donationPage/amountUtils.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/**
+ * Returns the index of an amount in a list of frequency page. If none can be
+ * found, this returns -1.
+ * @param {Object} page page object
+ * @param {number | string} amount to search for
+ * @param {string} frequency frequency code
```
Comment (GuiMend) [inline spa/src/components/donationPage/amountUtils.js:6]:
```suggestion
 * @param {string} frequency code
```

Comment (nrh-cklimas) [reply spa/src/components/donationPage/amountUtils.js]: Updated in 44de250ff3382046a192c6d6bd05300da44d8cf7

File: spa/src/components/donationPage/amountUtils.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+/**
+ * Returns the index of an amount in a list of frequency page. If none can be
+ * found, this returns -1.
+ * @param {Object} page page object
```
[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/src/components/donationPage/amountUtils.js]: Updated in 44de250ff3382046a192c6d6bd05300da44d8cf7

**Review by x110dc — APPROVED** [review]

**Pull request merged**